packages/bloc_test/lib/src/bloc_test.dart

```diff
@@ -170,6 +171,271 @@ void blocTest<B extends BlocBase<State>, State>(
   );
 }
 
+/// Creates a new `bloc`-specific test case with the given [description].
+/// [fakeAsyncBlocTest] will handle asserting that the `bloc` emits the
+/// [expect]ed states (in order) after [act] is executed.
+///
+/// The main difference of [fakeAsyncBlocTest] from [blocTest] is that [setUp],
+/// [act] and [tearDown] 'Functions' has parameter of type [FakeAsync] which
+/// provide explicitly control Dart's notion of the "current time". When the
+/// time is advanced, FakeAsync fires all asynchronous events that are scheduled
+/// for that time period without actually needing the test to wait for real time
+/// to elapse.
+///
+/// [fakeAsyncBlocTest] also handles ensuring that no additional states are
+/// emitted by closing the `bloc` stream before evaluating the [expect]ation.
+///
+/// [setUp] is optional and should be used to set up
```

Comment:
There is also some docs shared between `blocTest` and `fakeAsyncBlocTest`. We might wanna consider using dart doc template/macro to share a little bit of docs between them as well.